### PR TITLE
chore(node-ui): S3 polish — vocab + cross-layer pills + mini-graph colors + dedupe

### PR DESCRIPTION
**Pull request by Jurij89**
## Summary

Single polish PR bundling 10 of the 11 manual-test items from PR #772's review (item #5 deferred to Track A / S5 per ux-lead). Each item is a per-finding commit; lead-citation tags in the commit messages and below.

**Behaviour fixes**
- **#4** User-facing `"Verified"` → `"Verifiable"` vocab sweep (6 sites). Code identifiers (`TrustLevel = 'verified'`, `TRUST_COLORS.verified`) stay untouched per ux-lead's keep-list. _(ux-locked §1.5)_
- **#6** Cross-layer strip cells become interactive `<button aria-pressed=...>` wired to `toggleLayer`; `→` arrows stay inert. Inherits the "refuse last enabled" safeguard from `toggleLayer`. _(ui-locked)_
- **#7** `All` chip — extended tooltip clarifying the distinct-total math; inline `All = Root + subgraphs` hint (muted 11px, single line) after the Root chip. _(ux-locked option ii)_
- **#8** SWM-attribution discoverability — `Colored by contributing agent` badge above the Graph pane when narrowed to SWM-only; context-sensitive tooltip on the SWM cell. _(ux-locked a + c, b rejected)_
- **#9** `activeLayer` preservation on subgraph-chip navigation. `SubGraphBar.onSelect` carries `originatingLayer` as a 2nd arg; `ProjectView` plumbs it to `SubGraphDetailView.initialLayer`; that seeds `enabledLayers` to the originating layer instead of defaulting to all-three. Closes the §4.4.1 fold-in #4 contract. _(ux-locked §3 + §4.4.1 fold-in #4)_

**Visual polish**
- **#1** Mechanical rename `MiniLayerPyramid` → `MiniLayerBar`; `.v10-minipyr*` → `.v10-minibar*`. No visual change. _(ui-locked option b)_
- **#2 + #10b** `compact={true}` on both consumer mounts collapses MiniLayerBar's empty branch to `null`, eliminating the duplicate "No data" labels (card grid + detail-header corner). Card-body copy override: `entityCount > 0 ? 'Promoted — open to view' : 'No data yet'` (was `'No WM triples · promoted data only'` / `'No data yet'`). _(ui-locked compact, ux-locked card-body copy)_
- **#3** Per-trust mini-graph `nodeColors` plumbed via `graphOptions.style.nodeColors`. ui-locked priority chain: per-URI TRUST_COLORS wins → `defaultNodeColor` (card chrome) → namespaceColors. _(ui-locked)_
- **#10a** `.v10-subgraph-detail-title-wrap` reserves `min-height: 44px`; 1- and 2-line subgraph titles occupy the same vertical slot, no layout shift on navigation. _(ui-locked)_
- **#11** Cross-layer strip / lede / pill breathing room — `padding: 16px 16px 0` / `margin-bottom: 10px` / `margin-top: 14px`. Numeric deltas only, no new tokens, no DOM. _(ui-locked)_

**No-work**
- **#5** Partial breadcrumb ships as-is until Track A / S5 lands the deep multi-hop breadcrumb model. _(ux-locked)_

## Related

- Follows #772 (parent — S3 Subgraph Explorer)
- Track A S5 will carry #5 forward (Task #7)
- GH #771 / GH #773 are separate follow-ups, not in this PR
- Out-of-scope: Task #19 (`singleLayerPanelTriples` exact-tag-routing parallel construct)

## Files changed

| File | What |
|------|------|
| `packages/node-ui/src/ui/components/SubGraphBar.tsx` | #7 hint + tooltip on All chip; #9 `onSelect` 2-arg signature forwarding `originatingLayer` to every chip click |
| `packages/node-ui/src/ui/views/ProjectView.tsx` | #9 `subGraphInitialLayer` state + `handleSelectSubGraph` arg + `SubGraphDetailView.initialLayer` plumbing |
| `packages/node-ui/src/ui/views/project/components.tsx` | #1 component + mount rename; #2/#10b `compact` mounts + card-body copy; #3 `entityTrustByUriBySubGraph` derivation + `SubGraphMiniCard` per-URI nodeColors; #4 user-facing label rename (4 sites); #6 cross-layer cells become buttons + `aria-pressed`; #8 SWM-attribution badge + cell tooltip; #9 `initialLayer` prop + `initialEnabledLayers` seed |
| `packages/node-ui/src/ui/views/project/helpers.ts` | #4 `LAYER_CONFIG.vm.trustLabel` rename |
| `packages/node-ui/src/ui/styles.css` | #1 `.v10-minipyr*` → `.v10-minibar*` sweep; #6 cell hover / `aria-pressed` rules + `--bg-hover`; #7 `.v10-subgraph-bar-hint`; #8 attribution badge styles (reuses `--layer-shared` token); #10a title-wrap `min-height`; #11 spacing deltas |
| `packages/node-ui/test/context-graph-empty-stat-components.test.ts` | #4 vocab assertion update |
| `packages/node-ui/test/context-graph-ia-overview.test.ts` | #4 — legacy "Verified Memory" leak guard removed (moot post-sweep) |
| `packages/node-ui/test/sub-graph-bar-layer-scope.test.ts` | #7 hint + tooltip cases; #9 onSelect arg-forwarding cases (4 total new + 1 updated) |
| `packages/node-ui/test/subgraph-detail-view.test.ts` | #1 class-name updates; #2/#10b symmetric detail-header empty case; #6 cell click + safeguard; #8 SWM badge presence/absence; #9 `initialLayer` seeding (6 total new) |
| `packages/node-ui/test/subgraph-overview-grid.test.ts` | #2/#10b card-body two-branch cases + duplicate-label regression; #3 per-URI nodeColors + chrome-fallback regression (5 total new) |
| `packages/node-ui/test/ui-compat.test.ts` | #4 source-line guard update + test-description rename (inside `describe.skip`) |

## Test plan

- [x] `tsc --noEmit` exit 0
- [x] `pnpm --filter @origintrail-official/dkg-node-ui run build:ui` clean (only the existing chunk-size warning)
- [x] Extended S3 regression slice — **330 passed / 38 skipped** across 23 files, deterministic on warm rerun
  - Net new cases by item: #1 mechanical (no new); #2+#10b 4 new; #3 2 new; #4 vocab (existing updated, 1 line removed); #6 2 new; #7 3 new; #8 2 new; #9 5 new (1 existing updated)
- [x] Pre-flight gates:
  - `agent-docs/` NOT in diff
  - No new raw hex colors (`#3` uses `TRUST_COLORS[trustLevel]`; `#8` SWM dot reuses existing `--layer-shared` token)
  - No new design tokens introduced (`#11` numeric deltas; `#6` reuses `--bg-hover`)
  - `RESERVED_SUB_GRAPH_SLUGS` stays `Set(['meta'])`
- [ ] (Reviewer manual walkthrough — happy path) navigate Subgraphs → click a chip from WM tab → assert detail view's Active layer pill reads "Working Memory" + cross-layer cells show only WM pressed (#9). Click cells to toggle; click the last enabled WM cell → assert it stays pressed (#6 safeguard). Narrow to SWM → assert "Colored by contributing agent" badge above graph + SWM cell tooltip (#8).

## Commit history (per-finding)

- `afe9f0b13` #10a + #11 spacing (ui-locked, CSS-only)
- `fc20b6dde` #9 layer preservation (ux-locked §4.4.1 fold-in #4)
- `2d89c2c31` #8 SWM-attribution discoverability (ux-locked a + c)
- `e46ed609c` #7 All count hint + tooltip (ux-locked option ii)
- `762b1b2cf` #6 interactive cross-layer cells (ui-locked)
- `355af0115` #3 per-trust nodeColors (ui-locked priority chain)
- `878b0ecba` #2 + #10b compact mounts + card-body copy (ui-locked + ux-locked)
- `36c9647ba` #4 Verifiable vocab sweep (ux-locked §1.5)
- `6ad1eaa6f` #1 MiniLayerPyramid → MiniLayerBar (ui-locked option b)

### CODE REVIEW

File: packages/node-ui/src/ui/components/SubGraphBar.tsx [hunk 1]
```diff
@@ -244,8 +253,18 @@ export const SubGraphBar: React.FC<SubGraphBarProps> = ({ contextGraphId, profil
...
         onClick={() => onSelect(null)}
-        title={`All subgraphs · ${totalEntities} entities${scopeSuffix}${layerLabel ? '' : ` · ${totalTriples} triples`}`}
+        /* #7 polish (ux-locked option ii) — long-form tooltip
+           clarifies the All math because `MemoryEntity.subGraphs`
+           is a Set: cross-membership entities count once in `All`
+           but may also appear under multiple chip totals, and
+           Root holds entities in no sub-graph at all. The shorter
+           inline hint after the Root chip carries the same
+           semantic at a glance. */
+        title={`Total entities — includes those in named subgraphs (some may belong to more than one), plus Root entities not in any subgraph · ${totalEntities} entities${scopeSuffix}${layerLabel ? '' : ` · ${totalTriples} triples`}`}
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/components/SubGraphBar.tsx]: Fixed in `f9da6f948` — agreed with the finding.

Branched the copy on `layer !== undefined`:

- **Layer mode** (WM/SWM/VM): unchanged tooltip / aria / inline hint (`All = Root + subgraphs`) — All includes Root, math holds.
- **Layer-agnostic mode** (Subgraph Explorer overview): tooltip + aria rephrased to *"Total entities in named subgraphs (some may belong to more than one). Root entities are counted separately on the Root chip."*; the inline `All = Root + subgraphs` hint is **hidden entirely** since the equation doesn't hold here. Root chip itself stays visible.

Rejected the alternative (change `entityScopedAllTotal` to include Root in layer-agnostic mode) — the existing locked semantic at `SubGraphBar.tsx:135-145` was deliberate (drillable umbrella, not CG total) and changing it would regress the user's mental model.

5 test cases on this fix:
- existing "renders the hint" / "does NOT render when no Root chip" updated to mount with `layer: 'wm'` (preserving the populated-path coverage)
- NEW: `does NOT render the hint in layer-agnostic mode (Codex sweep 1)` — Root chip stays, hint hides
- NEW: layer-mode tooltip mentions `plus Root entities`, NOT `counted separately`
- NEW: layer-agnostic tooltip mentions `counted separately on the Root chip`, NOT `plus Root entities`

332/332 on the extended S3 slice; typecheck clean.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 2]
```diff
@@ -403,10 +410,14 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
...
-  // we were already on one.
-  const handleSelectSubGraph = useCallback((slug: string | null) => {
+  // we were already on one. The optional `originatingLayer` carries the
+  // bar's `layer` prop through to the detail view so the user's
+  // existing layer scope follows them across the navigation (#9
+  // polish, fold-in #4).
+  const handleSelectSubGraph = useCallback((slug: string | null, originatingLayer?: MemoryLayerView) => {
     clearDetailOrigin();
     setActiveSubGraph(slug);
+    setSubGraphInitialLayer(slug ? (originatingLayer ?? null) : null);
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx]: Fixed in `f0a3b0dab` — confirmed the bug shape (detail-view internal `SubGraphBar` is layer-agnostic by design, so hop 2+ silently widened to all-three).

Discriminator added per your guidance — three call patterns now route distinctly:

1. **Exit** (`slug === null`) → `setSubGraphInitialLayer(null)`.
2. **Layer-mode entry** (`originatingLayer` defined) → set scope to that layer.
3. **Layer-agnostic entry** (`originatingLayer === undefined`):
   - From the Subgraph Explorer overview (`activeSubGraph` was null pre-click) → keep null (fresh entry → all-three).
   - From an already-scoped detail (`activeSubGraph` non-null pre-click) → **preserve** prior scope via `setSubGraphInitialLayer(prev => prev)`.

The functional setter reads the latest state at click time; `activeSubGraph` joined the deps to keep the overview-vs-detail discriminator fresh.

3 new tests in `project-view-navigation.test.ts`:
- `preserves subGraphInitialLayer on detail→detail navigation (Bug H load-bearing)` — WM → alpha-with-layer → demo → other; `data-initial-layer` reads `wm` through all hops
- `keeps initialLayer === undefined on fresh entry from Subgraph Explorer overview` (regression guard for overview path)
- `exit (slug === null) clears any prior initialLayer scope` (regression guard for exit path)

Mock infrastructure: `SubGraphBar` mock now surfaces `data-layer` and provides `select-subgraph-alpha-with-layer` (layer-mode click) + `select-subgraph-other` (additional hop chip); `SubGraphDetailView` mock surfaces `data-initial-layer`.

File: packages/node-ui/src/ui/views/project/components.tsx [hunk 3]
```diff
@@ -4367,9 +4435,19 @@ export function SubGraphDetailView({
...
-  const [enabledLayers, setEnabledLayers] = useState<Set<TrustLevel>>(
-    () => new Set<TrustLevel>(['working', 'shared', 'verified']),
-  );
+  // #9 polish — `initialLayer` (when supplied by the chip-row
+  // click) seeds `enabledLayers` to that single layer instead of
+  // the default all-three. Carrying the originating scope
+  // across the navigation transition is the §4.4.1 fold-in #4
+  // contract. The chip click's `originatingLayer` arg flows from
+  // SubGraphBar → ProjectView.handleSelectSubGraph → here.
+  const initialEnabledLayers = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/project/components.tsx:4499]: Fixed in `77794cd47` — agreed on the indistinguishable-from-filter framing.

Three sites updated to derive against `initialEnabledLayers` instead of the hard-coded all-three:

1. **`hasAnyFilter`** — was `enabledLayers.size < 3 || ...`; now `!isAtSeededScope || chipState.size > 0 || !!queryResults`. Reset button visibility is honest: hidden at seeded state (whether all-three or single-layer).
2. **`resetFilters`** — `setEnabledLayers(new Set(initialEnabledLayers))` (restores seeded scope, not all-three).
3. **Active-layer pill** — `onClick` restores `initialEnabledLayers`; `disabled` keys off `isAtSeededScope`; `aria-label` + `title` branch on `initialEnabledLayers.size === 3` to surface `"click to reset to all layers"` vs `"click to restore originating scope"`.

`isAtSeededScope` is a single derived predicate (Set-equality against `initialEnabledLayers`) so the three consumers stay in lockstep.

4 tests touched in `subgraph-detail-view.test.ts`:
- NEW: `active-layer pill click restores initialEnabledLayers when seeded to a single layer` (load-bearing)
- NEW: `Reset filters restores initialEnabledLayers when seeded AND when chip filters exist`
- NEW: `Reset filters and pill restore all-three when initialEnabledLayers is the default` (regression guard for the all-three path)
- UPDATED: existing `seeds enabledLayers to the originating layer when initialLayer is provided` flipped `pill.disabled` assertion from `false` to `true` (the honest post-fix affordance — pill shouldn't claim to be clickable at the seeded scope)

338/338 on the extended slice; typecheck clean.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 4]
```diff
@@ -403,13 +410,45 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
...
     clearDetailOrigin();
     setActiveSubGraph(slug);
+    if (slug === null) {
+      setSubGraphInitialLayer(null);
+    } else if (originatingLayer !== undefined) {
+      setSubGraphInitialLayer(originatingLayer);
+    } else {
+      // Layer-agnostic — preserve existing scope on detail→detail
+      // hops, fall to null on fresh entry from the overview.
+      setSubGraphInitialLayer(prev => (activeSubGraph !== null ? prev : null));
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx]: Fixed in `b2b9aec49` — confirmed the bug; the Bug H fix preserved the stale seed when it should have routed the user's current scope.

Went with **path (b2)** from team-lead's brief — widened the carrier type rather than collapsing multi-layer to undefined. Concretely:

**SubGraphDetailView** (`components.tsx`):
- New `initialEnabledLayers?: ReadonlySet<TrustLevel>` prop. Wins over the legacy `initialLayer` single-layer alias when both are set; both fall back to default all-three.
- New `onEnabledLayersChange?: (Set<TrustLevel>) => void` callback prop. A useEffect mirrors `enabledLayers` up to the parent on mount and every change.

**ProjectView**:
- State: `subGraphInitialLayer: MemoryLayerView | null` → `subGraphInitialEnabledLayers: ReadonlySet<TrustLevel> | null`.
- New `detailScopeRef` + `handleDetailEnabledLayersChange` ref-mirror so the chip-click handler reads the latest scope **synchronously** (not bound by useCallback dep capture timing).
- `handleSelectSubGraph` layer-agnostic branch: `prev` (stale seed) → `detailScopeRef.current` (live scope). Snapshotted into a fresh Set to avoid sharing the reference with the unmounting detail view's state.

**Bug H semantic preserved**: when the user hasn't changed their scope, `detailScopeRef.current` equals the seed, so the hop still carries the seed through.

**Multi-layer round-trips cleanly** because the carrier is a Set, not a single-layer alias. Codex's (b1) path that papered over multi-layer with `undefined` is explicitly avoided.

**Tests** (+7 net):

`project-view-navigation.test.ts` (+4):
- `detail→detail nav preserves USER'S CURRENT scope after widening, not the stale seed (Bug J load-bearing)` — WM-seeded entry → widen to WM+SWM → hop to demo → asserts `data-initial-enabled-layers === 'shared,working'`
- `detail→detail nav carries narrowed-to-VM scope through the hop (Bug J symmetric)` — switching to a different single layer
- `detail→detail nav when current scope === seed still preserves correctly (Bug J / Bug H regression guard)` — proves the simple case still works
- `detail→detail nav when user widens to all-three carries multi-layer scope through` — full-widen edge case

`subgraph-detail-view.test.ts` (+3):
- `seeds enabledLayers from initialEnabledLayers prop (multi-layer, Bug J)`
- `initialEnabledLayers wins over initialLayer when both are set (precedence)`
- `mirrors enabledLayers up via onEnabledLayersChange on mount and on user toggles` — locks the structural mirror contract

Mock infra extended: `SubGraphDetailView` mock surfaces both `data-initial-layer` (back-compat — derived from `initialEnabledLayers` when size === 1, so existing Bug H assertions still pass) and `data-initial-enabled-layers` (comma-joined trust levels). Mock honours `onEnabledLayersChange` via its own useEffect so detail→detail tests see a populated `detailScopeRef`. Three buttons added to drive user-initiated scope changes.

345/345 on the extended slice; typecheck clean.

File: packages/node-ui/src/ui/components/SubGraphBar.tsx [hunk 5]
```diff
@@ -295,6 +336,18 @@ export const SubGraphBar: React.FC<SubGraphBarProps> = ({ contextGraphId, profil
...
+      {showRootChip && inLayerMode && (
+        /* #7 polish (ux-locked). Inline arithmetic-style hint
+           after the Root chip — explains the All math at a
+           glance. Muted tone so it doesn't compete with the
+           chip row; single line, never wraps (CSS).
+           Gated on `inLayerMode` (PR #793 Codex sweep 1 — the
+           layer-agnostic All total excludes Root by design, so
+           the equation only holds on WM/SWM/VM pages). */
+        <span className="v10-subgraph-bar-hint" aria-hidden="true">
+          All = Root + subgraphs
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/components/SubGraphBar.tsx]: Fixed in `411a89b19` — confirmed the false-equation framing (cross-membership entities count once in the distinct All total, so any sum/equation copy contradicts that math).

ux-lead landed verdict: replace the arithmetic literal with the legend-style `+ Root, subgraphs`. Reads as "this row includes Root and the subgraph chips", NOT as arithmetic. The leading `+` mirrors set membership without making a sum claim; the comma list avoids a second `+` that would still read as a sum.

Other options ux-lead considered + rejected:
- **(a) Drop hint entirely** — the user's original Image #23 complaint was a visual-scan issue; tooltip-on-hover discovers too late. The inline hint catches confusion before it fires.
- **(c) Conditional render on no-cross-membership** — hint visibility becoming data-dependent reads as flaky chrome.
- `+ Root` alone — too terse, doesn't say what's also there.
- `Includes Root` — loses visual rhythm with the chip row.
- `All ⊇ Root, subgraphs` — too cute.

Two-tier discoverability preserved: the inline hint is the at-a-glance visual cue; the long-form All-chip tooltip carries the cross-membership disclosure (`"some may belong to more than one"`) for users who investigate.

Test updates:
- Existing assertion `expect(hint!.textContent).toContain('All = Root + subgraphs')` → `toContain('+ Root, subgraphs')`
- Added regression guard `expect(hint!.textContent).not.toContain('All =')` so any accidental copy revert is caught
- `it()` titles and describe comments updated to the new literal

Sweep-1 mode-tagged tests are unaffected — they assert on mutually-exclusive tooltip phrases, not the hint literal. Bug K gate (`inLayerMode`) carries through unchanged.

20/20 on the bar slice; typecheck clean.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 6]
```diff
@@ -403,13 +421,52 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
...
+  //           seed. Pre-Bug-J this preserved the original seed
+  //           (e.g. user enters from WM, widens to WM+SWM in detail,
+  //           hops to another subgraph → next detail silently
+  //           narrowed back to WM-only). Reading `detailScopeRef`
+  //           gives us the user's actual current scope at click time.
+  const handleSelectSubGraph = useCallback((slug: string | null, originatingLayer?: MemoryLayerView) => {
     clearDetailOrigin();
     setActiveSubGraph(slug);
+    if (slug === null) {
+      setSubGraphInitialEnabledLayers(null);
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx:545]: Fixed in `69cb183b7` — confirmed the race window (mirror useEffect fires post-paint; fast hops complete in the gap).

Applied your suggested fix shape: every branch of `handleSelectSubGraph` that updates the seed state now writes `detailScopeRef.current` synchronously to the same value (exit → null, layer-mode entry → Set([trustOf(layer)]), layer-agnostic detail→detail → snapshot, layer-agnostic fresh entry → null). The new detail view's mirror useEffect still overwrites `ref.current` once it mounts — usually with the same value — and subsequently with the user's scope changes (Bug J's capture path). The sync write just closes the pre-mount window.

**Adjacent fix found while testing** — `activeSubGraph` from the useCallback closure could ALSO go stale during the same race (the first click's `setActiveSubGraph` hasn't re-rendered before the second click fires; closure capture is render-cycle bound). Added `activeSubGraphRef` updated at the top of every `handleSelectSubGraph` call so the overview-vs-detail discriminator reads the latest identity synchronously. Removed `activeSubGraph` from the useCallback deps since we read from the ref.

3 new tests in `project-view-navigation.test.ts`:
- `WM-tab → A → B fast hop synchronously preserves the WM seed (Bug L load-bearing)` — two clicks in a single `act()` block; pre-fix the second click read the stale ref + stale closure and snapped to all-three
- `exit-then-immediate-entry clears scope synchronously (Bug L exit-branch guard)` — exit→Subgraphs→C lands at all-three, NOT the stale prior-detail scope
- `overview → A → B fast hop reads non-null ref on the layer-agnostic path (Bug L cold-start)` — proves the overview-path branch's sync clear is load-bearing

350/350 on the extended slice; typecheck clean.

File: packages/node-ui/src/ui/views/project/components.tsx [hunk 7]
```diff
@@ -4097,10 +4119,33 @@ export function SubGraphMiniCard({
...
+  //   3. `namespaceColors` — built-in graph-viz tints, neutralised
+  //      to `card.color` so cross-cutting namespaces don't drown
+  //      the per-trust signal.
+  // Build at the card scope from `card.entityTrustByUri` (attached
+  // by the parent `cards` derivation; same canonical key the
+  // entity-only filter uses, so a coloured node here matches the
+  // node the entity-only filter admitted on the canvas).
+  const nodeColors = useMemo(() => {
+    const out: Record<string, string> = {};
+    for (const [uri, trustLevel] of card.entityTrustByUri) {
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/project/components.tsx:4179]: Fixed in `be6f86abb` — applied the dual-key pattern Codex referenced.

Implemented at the **upstream** `entityTrustByUriBySubGraph` derivation (`components.tsx:3989`) rather than the consumer:
```ts
m.set(canonical, e.trustLevel);
if (canonical !== e.uri) m.set(e.uri, e.trustLevel);
```

Reason: the consumer `SubGraphMiniCard.nodeColors` (`:4150`) iterates the map and writes both keys into the override automatically — no further consumer-side change needed. Side benefit: any future consumer of `entityTrustByUri` (e.g. a hover-tooltip-by-trust feature) inherits the dual-key contract without re-implementing the pattern at each call site.

Pattern is consistent with `trustNodeColors` (~:4625) which Codex referenced as the model.

2 new tests in `subgraph-overview-grid.test.ts`:
- `mini-graph nodeColors maps both raw <urn:...> and canonical urn:... forms to TRUST_COLORS (Bug M load-bearing)` — asserts BOTH `'<urn:e:wrapped>'` AND `'urn:e:wrapped'` keys present with `TRUST_COLORS.shared`. A sibling unwrapped entity (`urn:e:other`) also asserts canonical-only resolution still works (regression guard against accidental write-order inversion).
- `mini-graph nodeColors keeps the canonical mapping for unwrapped entities (Bug M regression guard)` — proves the dual-key shape doesn't introduce a phantom `<urn:e:plain>` key when the entity didn't have a wrapped form.

350/350 on extended slice; typecheck clean.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 8]
```diff
@@ -125,6 +125,33 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
...
+  }, []);
+  // PR #793 sweep 4 Bug L — `activeSubGraph` from the useCallback
+  // closure can be stale during a fast chip-to-chip burst (the
+  // state update from click 1 hasn't re-rendered before click 2
+  // fires). A ref kept in sync at the top of every
+  // handleSelectSubGraph call captures the latest active-subgraph
+  // identity synchronously so the layer-agnostic branch's
+  // "overview-vs-detail" discriminator stays fresh across the
+  // race window.
+  const activeSubGraphRef = useRef<string | null>(null);
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx:176]: Fixed in `6c8d4aeee` — direct follow-on from the Bug L `activeSubGraphRef` catch.

Confirmed three call sites mutate `activeSubGraph`:
- `handleSelectSubGraph` — already synced refs via Bug L
- `handleLayerSwitch` (`:516`) — bypassed refs entirely
- `handleDetailClose` (`:551`) — bypassed refs entirely (M2 origin restore)

Applied team-lead's option (i) — a single `setActiveSubGraphSync` helper that writes both ref and state atomically, plus clears `detailScopeRef` on the null branch (exit always clears scope). All three call sites swept through the helper. The discriminator invariant is restored: refs always equal the state they're modeling, regardless of which mutation path fired.

Rejected option (ii) (derive discriminator from state in an effect) — would reintroduce the post-paint timing window that Bug L's ref-based discriminator was added to close.

Inside `handleSelectSubGraph` the helper supersedes the manual ref writes — branches now compute the seed-state value and `detailScopeRef.current`'s post-write value explicitly. Comment block updated to call out the dual-write contract.

3 new tests in `project-view-navigation.test.ts`:
- `subgraph A (WM-seeded) → layer switch → overview → click B is a FRESH overview entry (Bug N load-bearing)` — exactly the failure flow you described
- `subgraph A → layer switch → layer-mode chip click still seeds the new layer (regression guard)` — proves the helper's null-clear didn't accidentally break originating-layer carry-over for the next chip click
- `subgraph A → open entity → detail-close → click B is a FRESH layer-tab entry (Bug N close-path)` — exercises the `handleDetailClose` call site round-trip

353/353 on extended slice; typecheck clean. CI will rerun on this push as you noted.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 9]
```diff
@@ -575,6 +718,8 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
             onSelectEntity={handleNavigate}
             activeTab={subGraphTabs[activeSubGraph] ?? 'items'}
             onTabChange={tab => handleSubGraphTabChange(activeSubGraph, tab)}
+            initialEnabledLayers={subGraphInitialEnabledLayers ?? undefined}
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx:797]: Fixed in `d857afabe` — confirmed the same-screen disagreement.

Went with **path (b)** — widened the bar's scope prop to a `Set<TrustLevel>` carrier mirroring Bug J's (b2). Cascade footprint was contained (3 derivations + the layerLabel + the inLayerMode discriminant inside SubGraphBar; one wiring change in ProjectView).

**SubGraphBar**:
- New `enabledScope?: ReadonlySet<TrustLevel>` prop. Internal `allowedTrustLevels` Set derives from `enabledScope` (wins when non-empty, collapses to layer-agnostic at size 3) or the legacy `layer` single-layer prop. `inLayerMode` discriminant becomes `allowedTrustLevels !== null`, generalising the #7 / Bug K layer-mode tooltip branching.
- Three derivations (`entityScopedCounts`, `entityScopedAllTotal`, `rootEntityCount`) all gate on `allowed.has(e.trustLevel)`.
- `layerLabel` for tooltips concatenates the present layer titles via `+` for multi-layer scopes.

**ProjectView**:
- New `currentDetailScope` state alongside the existing `detailScopeRef`. `handleDetailEnabledLayersChange` writes both in lockstep. `setActiveSubGraphSync` clears both on exit. `handleSelectSubGraph`'s three branches (layer-mode entry / detail→detail hop / overview-entry) set the state to match the ref.
- SubGraphBar mount at the activeSubGraph branch now passes `enabledScope={currentDetailScope ?? undefined}`.

8 new tests:
- `sub-graph-bar-layer-scope.test.ts` (+4): unit-level — `enabledScope=Set([working])` single-layer; `Set([working, shared])` multi-layer (pre-Bug-O unreachable); `Set([all-three])` collapses to layer-agnostic; precedence (`enabledScope` wins over `layer`)
- `project-view-navigation.test.ts` (+4): integration — single-layer entry / multi-layer widen-during-detail / overview-entry regression guard / exit clears both mirrors

364/364 on the extended slice; typecheck clean.

File: packages/node-ui/src/ui/views/ProjectView.tsx [hunk 10]
```diff
@@ -451,11 +588,17 @@ export function ProjectView({ contextGraphId }: ProjectViewProps) {
...
     if (!origin) return;
     setActiveLayer(origin.activeLayer);
-    setActiveSubGraph(origin.activeSubGraph);
+    // PR #793 sweep 5 Bug N — restoring the M2 origin's
+    // activeSubGraph must go through the sync helper so the refs
+    // mirror the restored state. When origin.activeSubGraph is
+    // null (entity opened from a layer-tab origin), the helper's
+    // null branch also clears detailScopeRef so the next chip
+    // click sees a clean overview-entry context.
+    setActiveSubGraphSync(origin.activeSubGraph);
```
[the remark this answers is not in the document]
Comment (Jurij89) [reply packages/node-ui/src/ui/views/ProjectView.tsx:654]: Fixed in `aafa71080` — confirmed the widen-then-snap-back flow.

Went with **option (ii)** — extended `DetailOrigin` with a `subGraphEnabledLayers: ReadonlySet<TrustLevel> | null` snapshot. Reason: option (i) (snapshot at entity-open into the seed state) would mutate `subGraphInitialEnabledLayers` mid-flight, blurring the "entry seed" semantic. Option (ii) keeps the original seed preserved as the entry baseline and makes the close-path restoration a single self-contained code section.

**captureDetailOrigin**: reads `detailScopeRef.current` (the sync source of truth at entity-open time) and snapshots into a fresh Set so the unmounting detail's state mutation can't share the ref. `null` when origin is a layer-tab (no subgraph context).

**handleDetailClose**: after the existing `setActiveSubGraphSync` restore, if origin had a subgraph scope, restores all three mirrors in lockstep:
```ts
setSubGraphInitialEnabledLayers(restored);
detailScopeRef.current = restored;
setCurrentDetailScope(restored);
```
The seed-state restore is what makes the remounted SubGraphDetailView honor the post-widening scope.

3 new tests in `project-view-navigation.test.ts`:
- `subgraph A widened to WM+SWM → open entity → close-back → scope stays at WM+SWM (Bug P load-bearing)` — drives the exact flow you described; pre-fix `data-initial-enabled-layers === 'working'`, post-fix `=== 'shared,working'`. Also asserts the sibling bar's `data-enabled-scope` matches the restored scope.
- `subgraph A at seed scope → open entity → close-back → scope still at seed (regression guard)`
- `entity opened from a layer-tab origin → close-back → no scope restoration (Bug P origin-shape guard)` — proves the null branch is preserved when origin has no subgraph context.

364/364 on extended slice; typecheck clean.

Function name ground-truth: the entity-open path is `handleNavigate` → `openEntityDetail` → `captureDetailOrigin`. The Bug P snapshot lives in `captureDetailOrigin` so it captures regardless of entry-point (graph node click, list click, navigation jump).
